### PR TITLE
Add an instrument status table to the main page

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-2106

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L316](https://github.com/TGAC/miso-lims/blob/a01fc4a8092367cd1272835d1ad13acfc08ef02c/miso-web/src/main/webapp/scripts/lims.js#L316): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L520](https://github.com/TGAC/miso-lims/blob/a01fc4a8092367cd1272835d1ad13acfc08ef02c/miso-web/src/main/webapp/scripts/lims.js#L520): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L618](https://github.com/TGAC/miso-lims/blob/a01fc4a8092367cd1272835d1ad13acfc08ef02c/miso-web/src/main/webapp/scripts/lims.js#L618): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by callunity** [comment]
Thoughts:
  * I'd prefer they were sorted by Platform (and secondarily by name)
  * Can we exclude the non-sequencing instruments since they'll never not be Idle?
  * Could you change the title to "Instrument Status" or "Sequencer Status"?
  * Could you bump the items per page up to 50? It took me a long time to realize there was one sequencer on the next page.
  * Instruments that last had a failed run probably shouldn't be displayed as `Status: Failed`. 
  * What do you think about deleting the model column? I'm assuming that everyone who's looking at this knows what kind of model each of these are.
  * Do we need columns for both run name and run alias?
  * Ultimately, I'd love it if this didn't take up the entire width of the page... but then we get back into CSS pain so this isn't a strong request.

**Comment by apmasell** [comment]
> Can we exclude the non-sequencing instruments since they'll never not be Idle?
They are excluded. Most of the instruments in the DB are mislabelled. We can just fix that data.
> Could you bump the items per page up to 50?
That's baked into a lot of the other table stuff, so I'd rather not change this. When we fix the non-sequencer instruments, it will fit.
> Instruments that last had a failed run probably shouldn't be displayed as Status: Failed.
Why not? This was intentional on my part. If the run failed on that sequencer, isn't that relevant to display?
> Do we need columns for both run name and run alias?
We've got them like that everywhere else, but I don't mind changing them.
> Ultimately, I'd love it if this didn't take up the entire width of the page
I think redesigning the rest of the main page is the answer on that one. If there was a search box that took the full width and a then the status that took the full width, that would be sensible.

**Comment by callunity** [comment]
Short answers: Yes to fixing the mislabelled instruments in the DB. Ok to not changing the items per page. Yes to redesigning the home page.

Longer:
>Instruments that last had a failed run probably shouldn't be displayed as Status: Failed.
>  Why not? This was intentional on my part. If the run failed on that sequencer, isn't that relevant to display?
In testing, SN7001205 is showing "Failed" for a run that started in February 2017. Since the sequencer is available once the run has completed, a sequencer status of "Failed" isn't accurate. ("Failed" is a run state, not a sequencer state).

(I think it would help to add an "Out of Service" status to instruments, but that's beyond the scope of this ticket.)

(I quietly hope that adding this status widget will get the lab to mark more instruments as Retired.)

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L316](https://github.com/TGAC/miso-lims/blob/a4030c6baac2f3e41bb3a571825391763decfab8/miso-web/src/main/webapp/scripts/lims.js#L316): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L520](https://github.com/TGAC/miso-lims/blob/a4030c6baac2f3e41bb3a571825391763decfab8/miso-web/src/main/webapp/scripts/lims.js#L520): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L618](https://github.com/TGAC/miso-lims/blob/a4030c6baac2f3e41bb3a571825391763decfab8/miso-web/src/main/webapp/scripts/lims.js#L618): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L316](https://github.com/TGAC/miso-lims/blob/2f9c102046184000df30bbc7bf22111152a60524/miso-web/src/main/webapp/scripts/lims.js#L316): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L520](https://github.com/TGAC/miso-lims/blob/2f9c102046184000df30bbc7bf22111152a60524/miso-web/src/main/webapp/scripts/lims.js#L520): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L618](https://github.com/TGAC/miso-lims/blob/2f9c102046184000df30bbc7bf22111152a60524/miso-web/src/main/webapp/scripts/lims.js#L618): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by djcooke** [comment]
Thoughts on the new (and much improved) look:
* Needs separators. Currently the instrument name is spaced more apart from its own related info than from the previous item's
* Can "Busy/Idle since..." float right beside the sequencer name instead of having its own line?
* Is the "Run: RUN123" line necessary, or can that space be saved if the alias is sufficient?
* Would an "@" before the cycle count help to clarify its meaning?

**Comment by djcooke** [comment]
* Should show hand cursor like links on mouseover to make it obvious that you can click the tiles
* This is slowing down the home page load significantly (over 8 seconds on my machine). I'm not sure what is causing that as the view seems really fast. Would be preferable to load the widget asynchronously either way

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TileTag.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  private String target;
+
+  @Override
+  protected int doStartTagInternal() throws Exception {
+    if (items == null) {
+      return SKIP_BODY;
+    }
+    @SuppressWarnings("unchecked")
+    Collection<Object> items = (Collection<Object>) this.items;
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TileTag.java]: MAJOR Rename "items" which hides the field declared at line 29. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TileTag.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    if (items.size() == 0 && !alwaysShow) {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TileTag.java]: MINOR Use isEmpty() to check whether the collection is empty or not. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TileTag.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+import org.springframework.web.servlet.tags.RequestContextAwareTag;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@SuppressWarnings("serial")
+public class TileTag extends RequestContextAwareTag {
+  private boolean alwaysShow;
+
+  private Object items;
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/TileTag.java]: CRITICAL Make "items" transient or serializable. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1948)

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/IlluminaRun.java [hunk 4]
```diff
@@ -98,4 +99,17 @@ public PlatformType getPlatformType() {
     return PlatformType.ILLUMINA;
   }
 
+  @Override
+  public String getProgress() {
+    if (getHealth() == HealthType.Running) {
+      if (numCycles != null && callCycle != null) {
+        if (numCycles != 0) {
```
Comment (sqwci) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/IlluminaRun.java]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)

File: miso-web/src/main/webapp/scripts/tile.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+TileTarget = {};
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/tile.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "TileTarget" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)

File: miso-web/src/main/webapp/scripts/tile.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+Tile = {
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/tile.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "Tile" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)

**Comment by sqwci** [comment]
SonarQube analysis reported 10 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [MenuController.java#L232](https://github.com/TGAC/miso-lims/blob/808c9a93160a803693605dca7fada98a6e4ff709/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/MenuController.java#L232): Refactor this method to reduce its Cognitive Complexity from 26 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [lims.js#L316](https://github.com/TGAC/miso-lims/blob/808c9a93160a803693605dca7fada98a6e4ff709/miso-web/src/main/webapp/scripts/lims.js#L316): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L520](https://github.com/TGAC/miso-lims/blob/808c9a93160a803693605dca7fada98a6e4ff709/miso-web/src/main/webapp/scripts/lims.js#L520): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L618](https://github.com/TGAC/miso-lims/blob/808c9a93160a803693605dca7fada98a6e4ff709/miso-web/src/main/webapp/scripts/lims.js#L618): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by apmasell** [comment]
I've made most of the changes.

> Is the "Run: RUN123" line necessary, or can that space be saved if the alias is sufficient?

I've gotten rid of `Run:`. Should I get rid of the actual run name? I was trying to follow the pattern everything else uses.

> Needs separators. Currently the instrument name is spaced more apart from its own related info than from the previous item's

I've change the size, but I intended there to be a gap between the instrument name and the rest.

> Can "Busy/Idle since..." float right beside the sequencer name instead of having its own line?

This is really hard to layout properly and looks very dense.

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/InstrumentStatusRestController.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+@RequestMapping("/rest/instrumentstatus")
+public class InstrumentStatusRestController extends RestController {
+  private static final Logger log = LoggerFactory.getLogger(InstrumentStatusRestController.class);
+
+  @Autowired
+  private InstrumentStatusService instrumentStatusService;
+
+  @RequestMapping(method = RequestMethod.GET, produces = "application/json")
+  @ResponseBody
+  public List<InstrumentStatusDto> listAll() throws IOException {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/InstrumentStatusRestController.java]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/InstrumentStatusRestController.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+import uk.ac.bbsrc.tgac.miso.core.data.type.HealthType;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.InstrumentStatusDto;
+import uk.ac.bbsrc.tgac.miso.service.InstrumentStatusService;
+
+@Controller
+@RequestMapping("/rest/instrumentstatus")
+public class InstrumentStatusRestController extends RestController {
+  private static final Logger log = LoggerFactory.getLogger(InstrumentStatusRestController.class);
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/InstrumentStatusRestController.java]: MAJOR Remove this unused "log" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

File: miso-web/src/main/webapp/scripts/tile.js [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    jQuery.ajax({
+      'dataType': 'json',
+      'type': 'GET',
+      'url': url,
+      'contentType': 'application/json; charset=utf8',
+      'success': function(data, textStatus, xhr) {
+        Tile.createPane(elementId, target, data);
+      },
+      'error': function(xhr, textStatus, errorThrown) {
+        var lines = ['Error: ' + errorThrown];
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/tile.js]: MINOR Remove the declaration of the unused 'lines' variable. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)

**Comment by sqwci** [comment]
SonarQube analysis reported 12 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L326](https://github.com/TGAC/miso-lims/blob/e840665f8328c67535c9e0500b6f2cdd2364efec/miso-web/src/main/webapp/scripts/lims.js#L326): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L532](https://github.com/TGAC/miso-lims/blob/e840665f8328c67535c9e0500b6f2cdd2364efec/miso-web/src/main/webapp/scripts/lims.js#L532): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L630](https://github.com/TGAC/miso-lims/blob/e840665f8328c67535c9e0500b6f2cdd2364efec/miso-web/src/main/webapp/scripts/lims.js#L630): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by callunity** [comment]
Could you please add "Lane #" to the `View [POOL_NAME]` links?

**Comment by apmasell** [comment]
Since the same pool can occur in multiple lanes, I deduplicate them, so that information isn't preserved.

**Comment by callunity** [comment]
Hmm. It's not necessarily obvious to me what's going on with the list of pools, but maybe the lab will have more context.

File: miso-web/src/main/webapp/scripts/tile.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+TileTarget = {};
+
+Tile = (function() {
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/tile.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "Tile" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)

File: miso-web/src/main/webapp/scripts/tile.js [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+      jQuery.ajax({
+        'dataType': 'json',
+        'type': 'GET',
+        'url': url,
+        'contentType': 'application/json; charset=utf8',
+        'success': function(data, textStatus, xhr) {
+          Tile.createPane(elementId, target, data);
+        },
+        'error': function(xhr, textStatus, errorThrown) {
+          var lines = ['Error: ' + errorThrown];
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/tile.js]: MINOR Remove the declaration of the unused 'lines' variable. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)

File: miso-web/src/main/webapp/scripts/tile.js [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+          try {
+            var responseObj = JSON.parse(xhr.responseText);
+            if (responseObj.detail) {
+              var errorMessage = document.createElement("P");
+              errorMessage.innerText = responseObj.detail;
+              showPane(elementId, target, errorMessage);
+            }
+          } catch (e) {
+            // If we got detail, great; if we didn't meh.
```
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/tile.js]: show generic "something went wrong" error

**Comment by sqwci** [comment]
SonarQube analysis reported 11 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L326](https://github.com/TGAC/miso-lims/blob/6d4d63b74802800b0fb1f7035e600648d380959e/miso-web/src/main/webapp/scripts/lims.js#L326): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L532](https://github.com/TGAC/miso-lims/blob/6d4d63b74802800b0fb1f7035e600648d380959e/miso-web/src/main/webapp/scripts/lims.js#L532): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L630](https://github.com/TGAC/miso-lims/blob/6d4d63b74802800b0fb1f7035e600648d380959e/miso-web/src/main/webapp/scripts/lims.js#L630): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 11 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L326](https://github.com/TGAC/miso-lims/blob/6a30e284506ce7d8116c1f123a429e5e5b4738d9/miso-web/src/main/webapp/scripts/lims.js#L326): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L532](https://github.com/TGAC/miso-lims/blob/6a30e284506ce7d8116c1f123a429e5e5b4738d9/miso-web/src/main/webapp/scripts/lims.js#L532): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L630](https://github.com/TGAC/miso-lims/blob/6a30e284506ce7d8116c1f123a429e5e5b4738d9/miso-web/src/main/webapp/scripts/lims.js#L630): Remove this usage of confirm(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**